module2/src/com/epam/arraysofarrays/MatrixBuilder.java

```diff
@@ -0,0 +1,62 @@
+package com.epam.arraysofarrays;
+
+public class MatrixBuilder {
+    //task 4 notFixed
+    public static  int[][] buildMatrixAccordingToPattern(int n) {
+        int[][] matrix = new int[n][n];
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                if (i % 2 ==0) {
+                    matrix[i][j] = j + 1;
+                } else {
+                    matrix[i][j] = n - j;
+                }
+            }
+        }
+        return matrix;
+    }
+    // task 5
+    public static int[][] buildMatrixForTaskFive(int n) {
+        int[][] matrix = new int[n][n];
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                if (i + j < n) {
+                    matrix[i][j] = i + 1;
+                }
+            }
+        }
+        return matrix;
+    }
+    // task 6
+    public static int[][] buildMatrixForTaskSix(int n) {
+        int[][] matrix = new int[n][n];
+        int halfMatrixLength = n / 2;
+        final int rowBeforeMiddle = halfMatrixLength - 1;
```

Comment:
Delete halfMatrixLength, too many variables
```suggestion
        final int rowBeforeMiddle = n / 2 - 1;
```